service/src/integrationTest/kotlin/fi/espoo/evaka/serviceneed/ServiceNeedQueriesTest.kt

```diff
@@ -14,4 +16,15 @@ class ServiceNeedQueriesTest : PureJdbiTest(resetDbBeforeEach = true) {
     fun getServiceNeedOptionPublicInfos() {
         assertThat(db.read { tx -> tx.getServiceNeedOptionPublicInfos(PlacementType.values().toList()) }).isEmpty()
     }
+
+    @Test
+    fun getOnlyShownServiceNeedOptionPublicInfos() {
+        db.transaction { tx -> tx.insertServiceNeedOptions() }
+        db.transaction { tx -> tx.updateShowForCitizen() }
+        assertThat(db.read { tx -> tx.getServiceNeedOptionPublicInfos(PlacementType.values().toList()) }).hasSize(10)
```

Comment:
This test is a bit fragile since nothing guarantees that the amount of service need options stays the same. I would just check that the updated service need options is not returned by the function. You can also just use the service need option's id in the update query and the check.

Reply:
Agreed, even though in principle nothing completely protects the test from the test data being changed under it. Couldn't see how to use the option's ID since it's randomized in the test data, so I used their names instead.

Reply:
Oh, right, Monday morning... Done.